File: PODM/SW/pod-manager/podm-business-entities/src/main/java/com/intel/podm/business/entities/redfish/embeddables/TrustedModule.java
/*
 * Copyright (c) 2017 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.intel.podm.business.entities.redfish.embeddables;

import com.intel.podm.common.types.InterfaceType;
import com.intel.podm.common.types.InterfaceTypeSelection;
import com.intel.podm.common.types.Status;
import org.apache.commons.lang3.builder.EqualsBuilder;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.Enumerated;
import java.util.Objects;

import static javax.persistence.EnumType.STRING;

@Embeddable
public class TrustedModule {
    @Column(name = "firmware_version")
    private String firmwareVersion;

    @Column(name = "firmware_version_2")
    private String firmwareVersion2;

    @Column(name = "interface_type")
    @Enumerated(STRING)
    private InterfaceType interfaceType;

    @Column(name = "interface_type_selection")
    @Enumerated(STRING)
    private InterfaceTypeSelection interfaceTypeSelection;

    @Column(name = "status")
    private Status status;

    public String getFirmwareVersion() {
        return firmwareVersion;
    }

    public void setFirmwareVersion(String firmwareVersion) {
        this.firmwareVersion = firmwareVersion;
    }

    public String getFirmwareVersion2() {
        return firmwareVersion2;
    }

    public void setFirmwareVersion2(String firmwareVersion2) {
        this.firmwareVersion2 = firmwareVersion2;
    }

    public InterfaceType getInterfaceType() {
        return interfaceType;
    }

    public void setInterfaceType(InterfaceType interfaceType) {
        this.interfaceType = interfaceType;
    }

    public InterfaceTypeSelection getInterfaceTypeSelection() {
        return interfaceTypeSelection;
    }

    public void setInterfaceTypeSelection(InterfaceTypeSelection interfaceTypeSelection) {
        this.interfaceTypeSelection = interfaceTypeSelection;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }

        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        TrustedModule that = (TrustedModule) o;
        return new EqualsBuilder()
            .append(firmwareVersion, that.firmwareVersion)
            .append(firmwareVersion2, that.firmwareVersion2)
            .append(interfaceType, that.interfaceType)
            .append(interfaceTypeSelection, that.interfaceTypeSelection)
            .append(status, that.status)
            .isEquals();
    }

    @Override
    public int hashCode() {
        return Objects.hash(firmwareVersion, firmwareVersion2, interfaceType, status, interfaceTypeSelection);
    }
}

File: PSME/common/agent-framework/src/module/model/trusted_module.cpp
/*!
 * @copyright
 * Copyright (c) 2017-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * */

#include <agent-framework/module/constants/compute.hpp>
#include "agent-framework/module/model/trusted_module.hpp"


using namespace agent_framework::model;


const enums::Component TrustedModule::component = enums::Component::TrustedModule;
const enums::CollectionName TrustedModule::collection_name = enums::CollectionName::TrustedModules;

TrustedModule::TrustedModule(const std::string& parent_uuid, enums::Component parent_type) :
    Resource{parent_uuid, parent_type} {}


TrustedModule::~TrustedModule() {}


json::Json TrustedModule::to_json() const {
    json::Json json = json::Json();
    json[literals::Status::STATUS] = get_status().to_json();
    json[literals::Oem::OEM] = get_oem().to_json();
    json[literals::TrustedModule::INTERFACE_TYPE] = get_interface_type();
    json[literals::TrustedModule::FIRMWARE_VERSION] = get_firmware_version();
    return json;
}


TrustedModule TrustedModule::from_json(const json::Json& json) {
    TrustedModule trusted_module{};
    trusted_module.set_status(attribute::Status::from_json(json[literals::Status::STATUS]));
    trusted_module.set_oem(attribute::Oem::from_json(json[literals::Oem::OEM]));
    trusted_module.set_interface_type(json[literals::TrustedModule::INTERFACE_TYPE]);
    trusted_module.set_firmware_version(json[literals::TrustedModule::FIRMWARE_VERSION]);

    return trusted_module;
}
